contracts/oracle/DynamicVoltSystemOracle.sol

```diff
@@ -0,0 +1,216 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.13;
+
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+import {PCVOracle} from "./PCVOracle.sol";
+import {CoreRefV2} from "./../refs/CoreRefV2.sol";
+import {DynamicVoltRateModel} from "./DynamicVoltRateModel.sol";
+
+/// @notice contract that linearly interpolates VOLT rate over a year
+/// into the VOLT price after the oracle start time.
+/// The VOLT rate is dynamic and depends on a DynamicVoltRateModel and
+/// the current ratio of liquid reserves in the system.
+/// When PCV allocations change, this contract is notified and the oracle
+/// start time updates, creating a new interpolation over 1 year.
+/// @author Eswak, Elliot Friedman
+contract DynamicVoltSystemOracle is CoreRefV2 {
+    /// ------------- Events ---------------
+
+    /// @notice Event emitted when the Volt system oracle compounds.
+    /// Emits the end time of the period that completed and the new oracle price.
+    event InterestCompounded(
+        uint64 periodStartTime,
+        uint192 periodStartOraclePrice
+    );
+
+    /// @notice Event emitted when the Volt system oracle base rate updates.
+    event BaseRateUpdated(
+        uint256 periodStart,
+        uint256 oldRate,
+        uint256 newRate
+    );
+
+    /// @notice Event emitted when the Volt system oracle actual rate updates.
+    event ActualRateUpdated(
+        uint256 periodStart,
+        uint256 oldRate,
+        uint256 newRate
+    );
+
+    /// @notice Event emitted when reference to the PCV Oracle updates.
+    event PCVOracleUpdated(
+        uint256 blockTime,
+        address oldOracle,
+        address newOracle
+    );
+
+    /// @notice Event emitted when reference to the Rate Model updates.
+    event RateModelUpdated(
+        uint256 blockTime,
+        address oldRateModel,
+        address newRateModel
+    );
```

Comment:
This should be moved to `IDynamicVoltSystemOracle` interface

Reply:
Added an IDynamicVoltSystemOracle, I also imported IOracleV2 to make this interface an IOracleV2 (and added the read() method in the system oracle, because it was missing)